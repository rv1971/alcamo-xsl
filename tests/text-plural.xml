<?xml version="1.0" encoding="UTF-8"?>

<?xml-stylesheet href="#xslt" type="text/xml"?>

<a:doc xmlns:a="tag:rv1971@web.de,2021:alcamo-xsl#">
  <xsl:stylesheet
      xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
      xml:id="xslt"
      version="1.0">
    <xsl:import href="../xsl/text.xsl"/>

    <xsl:template match="/">
      <a:result>
        <xsl:apply-templates select="/a:doc/a:data"/>
      </a:result>
    </xsl:template>

    <xsl:template match="a:item">
      <a:item>
        <xsl:call-template select="." name="a:plural">
          <xsl:with-param name="singular" select="'cow'"/>
        </xsl:call-template>

        <xsl:value-of select="', '"/>

        <xsl:call-template select="." name="a:plural">
          <xsl:with-param name="singular" select="'ox'"/>
          <xsl:with-param name="plural" select="'oxen'"/>
        </xsl:call-template>
      </a:item>
    </xsl:template>
  </xsl:stylesheet>

  <a:data>
    <a:item>0</a:item>
    <a:item>0001</a:item>
    <a:item>2</a:item>
    <a:item>32</a:item>
  </a:data>
</a:doc>
