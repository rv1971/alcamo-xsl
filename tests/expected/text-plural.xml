<?xml version="1.0"?>
<a:result xmlns:a="tag:rv1971@web.de,2021:alcamo-xsl#">
  <a:item>0 cows, 0 oxen</a:item>
  <a:item>1 cow, 1 ox</a:item>
  <a:item>2 cows, 2 oxen</a:item>
  <a:item>32 cows, 32 oxen</a:item>
</a:result>
